core/App/contexts/store.tsx

```diff
@@ -1,53 +1,96 @@
 import React, { createContext, Dispatch, useContext, useReducer } from 'react'
 
-import { State } from '../types/state'
-
-import reducer, { ReducerAction } from './reducers/store'
-
-export const initialStateFactory = (): State => {
-  return {
-    onboarding: {
-      didAgreeToTerms: false,
-      didCompleteTutorial: false,
-      didCreatePIN: false,
-      didConsiderBiometry: false,
-    },
-    authentication: {
-      didAuthenticate: false,
-    },
-    loginAttempt: {
-      loginAttempts: 0,
-      servedPenalty: true,
-    },
-    lockout: {
-      displayNotification: false,
-    },
-    privacy: {
-      didShowCameraDisclosure: false,
-    },
-    preferences: {
-      useBiometry: false,
-    },
-    credential: {
-      revoked: new Set(),
-      revokedMessageDismissed: new Set(),
-    },
-    error: null,
-    loading: false,
+import { BifoldError } from '../types/error'
+import {
+  Authentication,
+  Credential,
+  Lockout,
+  LoginAttempt,
+  Onboarding,
+  Preferences,
+  Privacy,
+  State,
+} from '../types/state'
+
+import _defaultReducer, { ReducerAction } from './reducers/store'
+
+type Reducer = <S extends State>(state: S, action: ReducerAction) => S
+
+interface StoreProviderProps {
+  initialState?: State
+  reducer?: Reducer
+}
+
+export class DefaultBifoldState implements State {
```

Comment:
Im not sure if I agree with converting to classes. Interfaces don't actually get compiled which means smaller builds and there's no overhead of instantiating.

Reply:
Let me see if I can revert it without TS complaining too much.